modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/InlineIndexHelper.java

```diff
@@ -911,11 +931,19 @@ public int put(long pageAddr, int off, Value val, int maxSize) {
                 PageUtils.putLong(pageAddr, off + 1, val.getLong());
                 return size + 1;
 
-            case Value.FLOAT: {
+            case Value.FLOAT:
                 PageUtils.putByte(pageAddr, off, (byte)valType);
                 PageUtils.putInt(pageAddr, off + 1, Float.floatToIntBits(val.getFloat()));
                 return size + 1;
-            }
+
+            case Value.DECIMAL:
+                float fl = val.getFloat();
+                if (val.getBigDecimal().stripTrailingZeros().precision() > MAX_FLOAT_PRECISION)
```

Comment:
Did you test it on big values? 
Try to execute snippet below. Precision for the first value is 2, so it will be inlined. 
```Float.compare(BigDecimal.valueOf(610000000000.0).stripTrailingZeros().floatValue(), BigDecimal.valueOf(609999999998.0).floatValue())```